platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gui/GuiPlatformAccessor.java

```diff
@@ -185,40 +182,68 @@ public ShadowGraph getShadowGraph(@NonNull NodeId nodeId) {
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public int compareTo(@NonNull final EventOrderInfo that) {
+            if (this.consensusOrder == -1 && that.consensusOrder == -1) {
+                // neither have reached consensus
+                return this.timeReceived.compareTo(that.timeReceived);
+            } else if (this.consensusOrder == -1) {
+                // that has reached consensus but not this
+                return 1;
+            } else if (that.consensusOrder == -1) {
+                // this has reached consensus but not that
+                return -1;
+            } else {
+                // both have reached consensus
+                return Long.compare(this.consensusOrder, that.consensusOrder);
+            }
+        }
+    }
+
     /**
      * Get a sorted list of events.
      */
     public PlatformEvent[] getAllEvents(@NonNull NodeId nodeId) {
         Objects.requireNonNull(nodeId, "nodeId must not be null");
-        // There is currently a race condition that can cause an exception if event order changes at
-        // just the right moment. Since this is just a testing utility method and not used in production
-        // environments, we can just retry until we succeed.
-        int maxRetries = 100;
-        while (maxRetries-- > 0) {
-            try {
-                final EventImpl[] allEvents = getShadowGraph(nodeId).getAllEvents();
-                Arrays.sort(allEvents, (o1, o2) -> {
-                    if (o1.getConsensusOrder() != -1 && o2.getConsensusOrder() != -1) {
-                        // both are consensus
-                        return Long.compare(o1.getConsensusOrder(), o2.getConsensusOrder());
-                    } else if (o1.getConsensusTimestamp() == null && o2.getConsensusTimestamp() == null) {
-                        // neither are consensus
-                        return o1.getTimeReceived().compareTo(o2.getTimeReceived());
-                    } else {
-                        // one is consensus, the other is not
-                        if (o1.getConsensusTimestamp() == null) {
-                            return 1;
-                        } else {
-                            return -1;
-                        }
-                    }
-                });
-                return allEvents;
-            } catch (final IllegalArgumentException e) {
-                logger.error(EXCEPTION.getMarker(), "Exception while sorting events", e);
-            }
+
+        final EventImpl[] allEvents = getShadowGraph(nodeId).getAllEvents();
+
+        // If events reach consensus during the execution of this method, order will change.
+        // Capture information before sorting, since sorting breaks if order changes mid-sort.
+        final Map<Hash, EventOrderInfo> orderInfo = new HashMap<>();
+        for (final EventImpl event : allEvents) {
+            orderInfo.put(event.getHashedData().getHash(), EventOrderInfo.of(event));
```

Comment:
Are you intentionally using `getHashedData` here, but `getBaseEventHashedData` below?

Reply:
These two methods are aliases for the same thing. No idea why there are two completely different methods, but not really in the scope of this PR to remove one of them. I've changed my new code to only use `getHashedData()`, so at least this part of the code is self consistent.